libcomposefs/lcfs-mount.c

```diff
@@ -578,34 +578,40 @@ static errint_t lcfs_mount_erofs_ovl(struct lcfs_mount_state_s *state,
 	char imagemountbuf[] = "/tmp/.composefs.XXXXXX";
 	char *imagemount;
 	bool created_tmpdir = false;
-	char loopname[PATH_MAX];
+	char source[PATH_MAX];
 	int errsv;
 	errint_t err;
 	int loopfd;
 
 	image_flags = lcfs_u32_from_file(header->flags);
 
-	loopfd = setup_loopback(state->fd, state->image_path, loopname);
-	if (loopfd < 0)
-		return loopfd;
-
 	if (options->image_mountdir) {
 		imagemount = (char *)options->image_mountdir;
 	} else {
 		imagemount = mkdtemp(imagemountbuf);
 		if (imagemount == NULL) {
 			errsv = errno;
-			close(loopfd);
 			return -errsv;
 		}
 		created_tmpdir = true;
 	}
 
-	err = lcfs_mount_erofs(loopname, imagemount, image_flags, state);
-	close(loopfd);
+	sprintf(source, "/dev/fd/%d", state->fd);
+	err = lcfs_mount_erofs(source, imagemount, image_flags, state);
+
 	if (err < 0) {
```

Comment:
Please check this for the specific `errno` we're expecting if we do this on older kernels or ones with the "mount file-backed" feature disabled.  Probably something like `ENOTBLK` or so?